### PR TITLE
check for underscore for https

### PR DESCRIPTION
**Pull request by GretaP**
Add check for underscores for custom domain certificates

### CODE REVIEW

**Review by TooManyBees — DISMISSED** [review]

File: lib/github-pages-health-check/domain.rb [hunk 1]
```diff
@@ -449,8 +449,11 @@ def enforces_https?
       def https_eligible?
         # Can't have any IP's which aren't GitHub's present.
         return false if non_github_pages_ip_present?
-        # Must be a CNAME or point to our IPs.
 
+        # Can't have underscores in the domain name
+        return false if host.include?("_")
```
Comment (parkr) [inline lib/github-pages-health-check/domain.rb:454]: Should this invalidate the domain name as well, or just HTTPS eligibility? Does GitHub have many customer custom domains with underscores?

**Review by parkr — COMMENTED** [review]
Drive by comment! Hi!

File: lib/github-pages-health-check/domain.rb (hunk 1 above)
Comment (yoannchaudet) [reply lib/github-pages-health-check/domain.rb:454]: Hello there!

We do have people on HTTP only and with a `_` in their domain name unfortunately 🙈. Some are using Cloudflare which does issue certs, but Let's Encrypt does not.

File: lib/github-pages-health-check/domain.rb [hunk 2]
```diff
@@ -449,8 +449,11 @@ def enforces_https?
       def https_eligible?
         # Can't have any IP's which aren't GitHub's present.
         return false if non_github_pages_ip_present?
-        # Must be a CNAME or point to our IPs.
 
+        # Can't have underscores in the domain name
```
Comment (yoannchaudet) [inline lib/github-pages-health-check/domain.rb:453]:
```suggestion
        # Can't have underscores in the domain name (Let's Encyrpt does not allow it)
```

**Review by yoannchaudet — APPROVED** [review]
Looks good to me. That will matches the current state which is, if a domain name contains `_`, we are unable to obtain a TLS certificate from Let's Encrypt today.

File: lib/github-pages-health-check/domain.rb (hunk 1 above)
Comment (parkr) [reply lib/github-pages-health-check/domain.rb:454]: Gotta love the edge cases! 😄

Comment (yoannchaudet) [reply lib/github-pages-health-check/domain.rb:454]: We are full of them!

**Review by yoannchaudet — APPROVED** [review]

**Pull request merged**